traits/tests/test_ctraits.py

```diff
@@ -308,8 +308,33 @@ def _bar_changed(self, new):
         self.assertEqual(foo.bar_changed[0], "baz")
 
     def test_failed_attribute_access(self):
+        # Double-underscore names are special-cased.
+        non_dunder_names = [
+            "non_existent",
+            "__non_existent",
+            "non_existent__",
+            "_non_existent_",
+            "__a__b_",
+            "_",
+        ]
+
+        dunder_names = [
+            "__package__",
+            "__a__",
+            "____",
+            "___",
+            "__",
```

Comment:
should `__`, `___` and sisters (integer multiples of `_` in between two other `_`) really be considered as dundernames in Python? I assumed that dundernames were `__X__` where x can be an alphanumeric string of any length..

Reply:
Yes, there are some corner cases here that aren't clear. I went with the equivalent of `my_string.startswith("__") and my_string.endswith("__")` because it's easy to describe, understand and implement, and because it matches CPython's own choices with respect to string mangling of class variable names (`__a__` won't be mangled because it's a dunder name; `__a` _will_ be mangled because it's not, and for whatever reason, `__` is _not_ mangled, so it's treated like the dunder names).

In the worst case, we'll encounter a use of `__` that really needs lookups to return `None`. I doubt that will happen, and if it does turn out to be a real need we can update the code accordingly.